# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //drivers/adapter/khdf/linux git repository of OpenHarmony 3.1 codebase.

Clang 14 brings in a lot of new checks/warnings, and the current
codebase violates the declaration-after-statement check, so we need to
reduce that from an error to a warning in order to be able to compile
the code.

Signed-off-by: Esben Haabendal <esben.haabendal@huawei.com>
Upstream-Status: Inappropriate [code should be fixed instead]

diff --git a/drivers/adapter/khdf/linux/model/audio/Makefile b/drivers/adapter/khdf/linux/model/audio/Makefile
index cac77f148329..9a3045955065 100644
--- a/drivers/adapter/khdf/linux/model/audio/Makefile
+++ b/drivers/adapter/khdf/linux/model/audio/Makefile
@@ -20,6 +20,8 @@ KHDF_AUDIO_HI3516DV300_INC_DIR = drivers/hdf/framework/../peripheral/audio/chips
 KHDF_AUDIO_RK3568_DIR = $(KHDF_AUDIO_BASE_ROOT_DIR)/device/hihope/rk3568/audio_drivers
 KHDF_AUDIO_RK3568_INC_DIR = drivers/hdf/framework/../../device/hihope/rk3568/audio_drivers
 
+ccflags-y += "-Wno-error=declaration-after-statement"
+
 obj-$(CONFIG_DRIVERS_HDF_AUDIO) += \
           $(KHDF_AUDIO_ROOT_DIR)/core/src/audio_core.o \
           $(KHDF_AUDIO_ROOT_DIR)/core/src/audio_host.o \
